index.js

```diff
@@ -12,6 +14,11 @@ module.exports = (input, needle, replacement) => {
 	let ret = '';
 	let matchCount = 0;
 	let prevIndex = 0;
+	const startIndex = opts.fromIndex > 0 ? opts.fromIndex : 0;
+
+	if (startIndex > input.length) {
+		return input;
+	}
 
 	while (true) { // eslint-disable-line no-constant-condition
 		const index = input.indexOf(needle, prevIndex);
```

Comment:
The tests only prove the correctness of the output, not the implementation. I specifically wanted this option to improve efficiency in a certain situation. If you read the diff here like me, you'll see that there's no way for the `indexOf` call to use the `fromIndex` in the first iteration.